### ISSUE TITLE
Keybindings broken in VSCode web, MacOS

### ISSUE THREAD

**Issue by samdenty** [microsoft/vscode#163432]
The keybindings are broken on MacOS for "`" & "§". You can see this by going to vscode.dev on MacOS, and pressing both them keys

ScanCode 106 (106 in enum) is detected as `IntlBackslash` https://github.com/microsoft/vscode/blob/5e2f5e57cb2d3a222d43de51541b0eda207914e8/src/vs/base/common/keyCodes.ts#L334.

Using a scancode map is very error prone as all different platforms have different scancodes.

A possible fix is to use [`KeyboardEvent.key`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) instead of a scancode map.

**Comment by fny** [comment]
Hmm... this works for me, but I have a different issue. My keybindings from window tiling app stopped working entirely.

**Comment by samdenty** [comment]
I think that's because vscode intercepts all keyboard shortcuts, COMMAND+SHIFT+B to toggle bookmarks bar also doesn't work

**Comment by fny** [comment]
Just to add, I noticed this change after upgrading to the latest release.

**Comment by roblourens** [comment]
In case this helps https://github.com/microsoft/vscode/wiki/Keybinding-Issues#summary

**Comment by samdenty** [comment]
@roblourens is there a reason why it's broken in VSCode Web, but not on desktop? Setting `keyboard.dispatch` to `keyCode` fixes it, but it works with `code` on desktop

**Comment by alexdima** [comment]
> is there a reason why it's broken in VSCode Web

Yes, on the desktop we have a native node module which we use to detect the key mappings. On the web, we need to rely on web APIs, which are incomplete.

**Comment by fny** [comment]
This is happening to me on desktop FWIW.

**Comment by MasterInQuestion** [comment]
Similar also happening on Android:
    |&#x2A;| 【F1】 ("Command Palette") outright not responding.
    |&#x2A;| 【Ctrl】 + 【A】 ("Select All") has no effect [1] while 【Ctrl】 + 【C】/【V】 ("Copy" / "Paste") would work.
    |&#x2A;| 【Tab】 could not be trapped. (causes keyboard navigation)
    |&#x2A;| And many more.
[1] Not happening in Chromium.

**Comment by alexdima** [comment]
@fny @MasterInQuestion Keybindings are OS, browser and keyboard layout dependent, which means your problems are distinct from the OP. Please open a separate issue if you still have keybindings problems.

**Comment by samdenty** [comment]
Chrome on MacOS with British layout

**Comment by samdenty** [comment]
On native macos, it's broken for me in Firefox & Chrome in `keyboard.dispatch: code` setting it to keycode fixes it

It works in safari though

**Comment by alexdima** [comment]
Issue for Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1755096

**Comment by deepak1556** [comment]
> Found a newer bug on Chromium's side: https://bugs.chromium.org/p/chromium/issues/detail?id=1296783

Awesome, this should help us remove the patch.

**Comment by samdenty** [comment]
Yep, I can confirm it's broken in both Firefox/Chrome with doing:

```
window.onkeydown = (e) => console.log("code:", e.code, "key:", e.key);
```

**Comment by MasterInQuestion** [comment]
[ Quote alexdima @ CE 2022-12-06 21:27:18 UTC:
[https://github<i>.</i>com/microsoft/vscode/issues/163432#issuecomment-1340028915](https://github.com/microsoft/vscode/issues/163432#issuecomment-1340028915)
    Keybindings are OS, browser and keyboard layout dependent, which means you are running into problems which are distinct from the problem reported by the OP. ]
<^>    Fundamentally caused by identical cause. (key binding in default configuration in general broken across browsers)

**Comment by feraleyebrows** [comment]
Still broken on Chrome almost a year later.
